lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -587,6 +589,23 @@ bool TypeChecker::typeCheckForEachBinding(DeclContext *dc, ForEachStmt *stmt) {
   if (!typeCheckExpression(target))
     return failed();
 
+  // check to see if the sequence expr is throwing (and async), if so require the stmt to have a try loc
+  if (stmt->getAwaitLoc().isValid()) {
+    auto Ty = sequence->getType();
+    if (Ty.isNull()) { return failed(); }
+    auto context = sequence->getType()->getNominalOrBoundGenericNominal();
+    if (!context) { return failed(); }
```

Comment:
The sequence type might be a generic parameter. Please add a test case and remove this bail-out